src/basefoc.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * Source code for Urban motor controller functionality
+ */
+
+#include "../include/main.h"
+extern Encoder encoder;
+
+//  BLDCMotor(int pp, (optional R, KV))
+//  - pp  - pole pair number
+//  - R   - phase resistance value - optional
+//  - KV  - motor KV rating [rpm/V] - optional
+// For Surpass C2216-880KV motor | 14 pole pairs | 0.108 Resistance | 880KV
+BLDCMotor motor = BLDCMotor(14);
+
+// For Koford 129H169T motor | 10 pole pairs | TBD Resistance | TBD
+// BLDCMotor motor = BLDCMotor(10);
+
+//  BLDCDriver3PWM( int phA, int phB, int phC, int enA, int enB, int enC )
+//  - phA, phB, phC - A,B,C phase pwm pins
+//  - enA, enB, enC - enable pin for each phase (optional)
+BLDCDriver3PWM driver = BLDCDriver3PWM(12, 13, 14);
+
+// InlineCurrentSensor constructor
+//  - shunt_resistor  - shunt resistor value
+//  - gain  - current-sense op-amp gain
+//  - phA   - A phase adc pin
+//  - phB   - B phase adc pin
+//  - phC   - C phase adc pin (optional - _NC if not using)
+InlineCurrentSense current_sense  = InlineCurrentSense(0.01, 50.0, A0, T1, T2);
+
+// commander communication instance
+Commander command = Commander(Serial);
+void onMotor(char* cmd){ command.motor(&motor, cmd); }
+
+void BaseFOC( void * pvParameters ) {
+
+  Serial.println("Base Ready");
+  motor.useMonitoring(Serial);
+
+  // link the motor to the sensor
+  motor.linkSensor(&encoder);
+
+  // power supply voltage [V]
+  driver.voltage_power_supply = 48;
```

Comment:
I decided to set the parameters to what they would be for comp in GitHub, and make sure to change them locally during testing, since they will be changing regularly. Pushing each time didn't seem like the best call.